wallet/node/Main.hs

```diff
@@ -0,0 +1,97 @@
+import           Pos.Ssc.Class       (SscParams)
+import           Pos.Ssc.GodTossing  (SscGodTossing)
+import           Pos.Ssc.NistBeacon  (SscNistBeacon)
+import           Pos.Ssc.SscAlgo     (SscAlgo (..))
+import           Pos.Util.UserSecret (usVss)
+import           Pos.WorkMode        (WorkMode)
+import           Pos.Web             (serveWebGT)
+import           Pos.Wallet.Web      (WalletWebMode, bracketWalletWS, bracketWalletWebDB,
+                                      runWRealMode, walletServeWebFull, walletServerOuts)
+
+import           Pos.Client.CLI.NodeOptions (SimpleNodeArgs (..))
+import           Pos.Client.CLI.Params (getSimpleNodeParams, gtSscParams)
+import           Pos.Client.CLI.Util (printFlags)
+
+import           NodeOptions (WalletNodeArgs(..), WalletArgs(..), getWalletNodeOptions)
+
+actionWithWallet :: SscParams SscGodTossing -> NodeParams -> WalletArgs -> Production ()
+actionWithWallet sscParams nodeParams wArgs@WalletArgs {..} =
+    bracketWalletWebDB walletDbPath walletRebuildDb $ \db ->
+        bracketWalletWS $ \conn -> do
+            bracketNodeResources nodeParams sscParams $ \nr@NodeResources {..} ->
+                runWRealMode
+                    db
+                    conn
+                    nr
+                    (runNode @SscGodTossing nr plugins)
+  where
+    convPlugins = (, mempty) . map (\act -> ActionSpec $ \__vI __sA -> act)
+    plugins :: ([WorkerSpec WalletWebMode], OutSpecs)
+    plugins = convPlugins (pluginsGT wArgs) <> walletProd wArgs
+
+walletProd :: WalletArgs -> ([WorkerSpec WalletWebMode], OutSpecs)
+walletProd WalletArgs {..} = first pure $ worker walletServerOuts $ \sendActions ->
+    walletServeWebFull
+        sendActions
+        walletDebug
+        walletPort
+        (Just walletTLSParams)
+
+pluginsGT ::
+    ( WorkMode SscGodTossing ctx m
+    , HasNodeContext SscGodTossing ctx
+    ) => WalletArgs -> [m ()]
+pluginsGT WalletArgs {..}
+    | enableWeb = [serveWebGT webPort (Just walletTLSParams)]
+    | otherwise = []
+
+action :: WalletNodeArgs -> Production ()
+action (WalletNodeArgs (snArgs@SimpleNodeArgs {..}) (wArgs@WalletArgs {..})) = do
+    systemStart <- CLI.getNodeSystemStart $ CLI.sysStart commonArgs
+    logInfo $ sformat ("System start time is " % shown) systemStart
+    t <- currentTime
+    logInfo $ sformat ("Current time is " % shown) (Timestamp t)
+    currentParams <- getSimpleNodeParams snArgs systemStart
+    putText $ "Running using " <> show (CLI.sscAlgo commonArgs)
+    putText $ "Wallet is enabled!"
+
+    let vssSK = fromJust $ npUserSecret currentParams ^. usVss
+    let gtParams = gtSscParams snArgs vssSK
+
+    let sscParams :: Either (SscParams SscNistBeacon) (SscParams SscGodTossing)
+        sscParams = bool (Left ()) (Right gtParams) (CLI.sscAlgo commonArgs == GodTossingAlgo)
+    case sscParams of
+        (Left _)     -> logError "Wallet does not support NIST beacon!"
```

Comment:
It would be good to remove ability to run wallet with nist beacon completely (remove `sscAlgo` option from wallet). It can be done after this PR is merged though.